ontopy/excelparser.py

```diff
@@ -29,118 +35,248 @@ def create_ontology_from_excel(  # pylint: disable=too-many-arguments
+
     onto.sync_python_names()
     with onto:
-        # loop through the rows until no more are added
-        new_loop = True
-        final_loop = False
-        while new_loop:
-            number_of_added_classes = 0
-            for _, row in data.iterrows():
+        remaining_rows = set(range(len(data)))
+        while remaining_rows:
+            added_rows = set()
+            for index in remaining_rows:
+                row = data.loc[index]
                 name = row["prefLabel"]
                 try:
-                    if isinstance(
-                        onto.get_by_label(name), owlready2.ThingClass
-                    ):
-                        continue
+                    onto.get_by_label(name)
+                except (ValueError, TypeError) as err:
+                    warnings.warn(
+                        f'Ignoring concept "{name}". '
+                        f'The following error was raised: "{err}"'
+                    )
+                    continue
                 except NoSuchLabelError:
                     pass
 
-                parent_names = str(row["subClassOf"]).split(";")
-
-                try:
-                    parents = [onto.get_by_label(pn) for pn in parent_names]
-                except NoSuchLabelError:
-                    if final_loop is True:
-                        parents = owlready2.Thing
-
-                        warnings.warn(
-                            "Missing at least one of the defined parents. "
-                            f"Concept: {name}; Defined parents: {parent_names}"
+                if name in onto:
+                    if not force:
+                        raise ExcelError(
+                            f'Concept "{name}" already in ontology'
                         )
-                        new_loop = False
+                    warnings.warn(
+                        f'Ignoring concept "{name}" since it is already in '
+                        "the ontology."
+                    )
+                    # What to do if we want to add info to this concept?
+                    # Should that be not allowed?
+                    # If it should be allowed the index has to be added to
+                    # added_rows
+                    continue
+
+                if pd.isna(row["subClassOf"]):
+                    if not force:
+                        raise ExcelError(f"{row[0]} has no subClassOf")
+                    parent_names = []  # Should be "owl:Thing"
+                else:
+                    parent_names = str(row["subClassOf"]).split(";")
+
+                parents = []
+                invalid_parent = False
+                for parent_name in parent_names:
+                    try:
+                        parent = onto.get_by_label(parent_name.strip())
+                    except (NoSuchLabelError, ValueError) as exc:
+                        if parent_name not in labels:
+                            if force:
```

Comment:
So `force` means more the likes of "Don't raise unless there's a proper internal error"?

Reply:
I tried to make it clearer in the documentation.